reporter/src/main/kotlin/reporters/NoticeReporter.kt

```diff
@@ -109,13 +109,13 @@ class NoticeReporter : Reporter {
             "This project contains or depends on third-party software components pursuant to the following licenses:\n"
         }
 
-        val noticeReport = if (postProcessingScript != null) {
-            PostProcessor(
+        val noticeReport = if (preProcessingScript != null) {
+            PreProcessor(
                 ortResult,
                 NoticeReport(listOf(header), licenseFindings, emptyList()),
                 copyrightGarbage,
                 licenseConfiguration
-            ).run(postProcessingScript)
+            ).run(preProcessingScript)
```

Comment:
I believe this code snippet here shown quite nicely why do I dot agree with the rationale in the commit message and continue to believe the script is a *post*-processor:

The processor / script operates on the NoticeReport, which equals the object created in line 120 below, but does some additional things with it, *after* the NoticeReport has been created.

Reply:
I think the name `NoticeReport` for the data class was an unfortunate choice. The class is not the notice report, but the output file generated by the reporter is. The data class is merely a view on the `OrtResult` that contains the data required to generate a notice report (and not even all data).
In the end the question is what `pre` and `post` relate to, and I'd rather relate it to the generation of the actual notice report, the output file, than the collection of data from the `OrtResult`.
This might be more obvious once I have implemented the changes for #2001. The reporter will thank work like this:

* Collect the default notice report data.
* Pre-process this data with the custom user script.
* Process the data into a simplified format used only for rendering. For this step two different implementations will exist.
* Render the output file.

Reply:
> I'd rather relate it to the generation of the actual notice report

I believe I agree with that rationale: A pre-processor should be something that modifies the input for generating the report, before the actual report is created. A post-processor should be something that modifies the "final" report to really finalize it, but that still operates *before* the report is written to a file.

Your mentioned steps above sound to me like this is how it will be like, correct?

Reply:
Yes.